src/traces/treemap/plot.js

```diff
@@ -103,6 +103,10 @@ function plotOne(gd, cd, element, transitionOpts) {
         return;
     }
 
+    if(!trace.pathbar.visible) {
+        selAncestors.remove();
+    }
```

Comment:
It's probably cleaner to move this below from 

https://github.com/plotly/plotly.js/blob/326cb04f9f1fbf1d0f8133b8e4853ca8b2dd98ee/src/traces/treemap/plot.js#L615-L634

```js
if(trace.pathbar.visible) {
  // ...
} else {
   selAncestors.remove();
}
```

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4516/commits/838d1c39c076afca631b42fdfe208fc052fe7434.